### PR TITLE
feature-hw-2

### PR DESCRIPTION
**Pull request by kristina-salnyk**
Refactor an existing application to use TypeScript

### CODE REVIEW

File: src/middlewares/validateBody.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import {ObjectSchema} from "joi";
+import {NextFunction, Request, Response} from "express";
+
+import MovieInput from "../interfaces/MovieInput";
+import ApiError from "../interfaces/ApiError";
+
+const validateBody = (schema: ObjectSchema<MovieInput>) => {
+  return (req: Request, _res: Response, next: NextFunction): void => {
+    if (Object.keys(req.body).length === 0) {
+      const e = new Error('Invalid input') as ApiError;
```
Comment (KryvDanylo) [inline src/middlewares/validateBody.ts]: You can create a class ApiError extended from Error and avoid using "as".

**Review by KryvDanylo — COMMENTED** [review]
Also, it would be great if you added prettier config

**Comment by kristina-salnyk** [comment]
> Also, it would be great if you added prettier config

Configured Prettier, and added general formatting rules. Additionally, configured pre-commit validation using Husky.

File: src/middlewares/validateBody.ts (hunk 1 above)
Comment (kristina-salnyk) [reply src/middlewares/validateBody.ts]: Yea, of course, changes have been added.

**Comment by kristina-salnyk** [comment]
@KryvDanylo, please check changes

File: src/classes/ApiError.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class ApiError extends Error {
```
Comment (KryvDanylo) [inline src/classes/ApiError.ts]: Good, but a better name for the folder would be errors.

**Review by KryvDanylo — CHANGES_REQUESTED** [review]

File: src/classes/ApiError.ts (hunk 2 above)
Comment (kristina-salnyk) [reply src/classes/ApiError.ts]: Fixed

**Review by KryvDanylo — APPROVED** [review]
Good

**Pull request merged**